tickets/src/main/java/no/unit/nva/publication/ticket/update/updateTicketAssigneeHandler.java

```diff
@@ -0,0 +1,139 @@
+        this(TicketService.defaultService(),
+             ResourceService.defaultService(), new DataCiteDoiClient(HttpClient.newHttpClient(),
+                                                                     SecretsReader.defaultSecretsManagerClient(),
+                                                                     new Environment().readEnv("API_HOST")));
+    }
+
+    public updateTicketAssigneeHandler(TicketService ticketService, ResourceService resourceService,
+                                       DoiClient doiClient) {
+        super(TicketDto.class);
+        this.ticketService = ticketService;
+        this.resourceService = resourceService;
+        this.doiClient = doiClient;
+    }
+
+    @Override
+    protected Void processInput(TicketDto input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        var ticketIdentifier = extractTicketIdentifierFromPath(requestInfo);
+        var ticket = ticketService.fetchTicketByIdentifier(ticketIdentifier);
+
+        if (userIsNotAuthorized(requestInfo, ticket)) {
+            throw new ForbiddenException();
+        }
+        if (ticket instanceof DoiRequest) {
+            doiTicketSideEffects(input, requestInfo);
+        }
+        ticketService.updateTicketAssignee(ticket, input.getAssignee());
+        return null;
+    }
+
+    @Override
+    protected Integer getSuccessStatusCode(TicketDto input, Void output) {
+        return HttpURLConnection.HTTP_ACCEPTED;
+    }
+
+    private static boolean userIsNotAuthorized(RequestInfo requestInfo, TicketEntry ticket)
+        throws UnauthorizedException {
+        return !(isAuthorizedToClaimTickets(requestInfo) && isUserFromSameCustomerAsTicket(requestInfo, ticket));
+    }
+
+    private static boolean isUserFromSameCustomerAsTicket(RequestInfo requestInfo, TicketEntry ticket)
+        throws UnauthorizedException {
+        return requestInfo.getCurrentCustomer().equals(ticket.getCustomerId());
+    }
+
+    private static boolean isAuthorizedToClaimTickets(RequestInfo requestInfo) {
+        return requestInfo.userIsAuthorized(AccessRight.APPROVE_DOI_REQUEST.toString());
+    }
+
+    private void doiTicketSideEffects(TicketDto input, final RequestInfo requestInfo)
+        throws NotFoundException, BadMethodException, BadGatewayException {
+        var assignee = input.getAssignee();
+        var publication = getPublication(requestInfo);
+        var ticketAssignee = UserInstance.fromPublication(publication);
+        if (ticketAssignee.getUser().equals(assignee)) {
+            findableDoiTicketSideEffects(requestInfo);
+        }
+    }
+
+    private void findableDoiTicketSideEffects(RequestInfo requestInfo)
+        throws NotFoundException, BadMethodException, BadGatewayException {
+        var publication = getPublication(requestInfo);
+        publicationSatisfiesDoiRequirements(publication);
+        createFindableDoiAndPersistDoiOnPublication(publication);
```

Comment:
Shouldn't the ticket get status "complete" if the publication has a findable DOI?